.github/actions/build-frontend/action.yml

```diff
@@ -0,0 +1,41 @@
+# action.yml
+name: 'Build Frontend'
+description: 'Build react website'
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Use Node.js ${{ matrix.node-version }} with yarn
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - run: yarn install
+      working-directory: frontend-react
+      shell: bash
+
+    - name: Build release for production branch (conditional check)
+      if: github.ref == 'refs/heads/production'
+      working-directory: frontend-react
+      run: yarn build:production
+      shell: bash
+
+    - name: Build staging main/master branch (conditional check)
+      if: (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/main')
+      working-directory: frontend-react
+      run: yarn build:staging
+      shell: bash
+
+    - name: Build staging fallthrough for PR branch (conditional check)
+      if: (!(github.ref == 'refs/heads/master' || github.ref == 'refs/heads/main' || github.ref == 'refs/heads/production'))
+      working-directory: frontend-react
+      run: yarn build:staging
```

Comment:
I guess we can combine this and the previous step into one, changing the conditional to:

```
      if: github.ref != 'refs/heads/production'
```

since the `run` step is the same for both. So we basically just have a 'production' build and a 'non-production' (aka, staging) build.

Reply:
That's true. I just copied exactly what they had in `build_frontend.yml` to help make the change less dramatic, but I can certainly improve the code.